src/oscar/apps/customer/migrations/0006_auto_20190430_1736.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 2.0.13 on 2019-04-30 16:36
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('customer', '0005_auto_20181115_1953'),
+    ]
+
+    state_operations = [
+        migrations.DeleteModel(
+            name='CommunicationEventType',
+        ),
+        migrations.RemoveField(
+            model_name='email',
+            name='user',
+        ),
+        migrations.RemoveField(
+            model_name='notification',
+            name='recipient',
+        ),
+        migrations.RemoveField(
+            model_name='notification',
+            name='sender',
+        ),
+        migrations.DeleteModel(
```

Comment:
Deleting data is something we need to be careful about. If we're going to do this I think we need to document very clearly that these models are being deleted. Given that many sites may want to migrate data, I actually wonder whether we need to stage this, so that we:

1. Add new models and allow projects to migrate data into them.
2. Deprecate the old models in Oscar 2.1
3. Delete the old models in Oscar 2.2.

Doing everything in one go will make it very difficult for projects to do data migrations.

Reply:
In general, we do not delete data here, we just mark these models as *deleted* (with `state_operations` no DB changes will be made).

The same we are doing in the `communications/migrations/0001_initial.py` (https://github.com/django-oscar/django-oscar/pull/2591/files#diff-c64e3a7f51c8fb93665d85a22e77896dR21) - no new DB tables for the `communication` app will be created - we only mark them as created, but the tables from the `customer` app will be used.

Please correct me if I missed something.

Reply:
I'd missed that these were state operations - thanks for clarifying!

Please ping me once the rest of the PR is ready for another review.